### PR TITLE
Rename SCOPE_ROOT, ${@}, ${^}, etc.

### PR DESCRIPTION
**Pull request by rgrinberg**

### CODE REVIEW

**Comment by rgrinberg** [comment]
The code duplication is temporary as I have a plan to clean this up after this and #956 are merged.

**Comment by emillon** [comment]
I started to work on this, but for now I only had some test cases, so I'll open a separate PR for these.

**Comment by emillon** [comment]
No problem! I just added the test case in question.

**Comment by rgrinberg** [comment]
Btw @diml I ended up writing a custom lexer for parsing `deps[x]` because making it unavailable in bootstrap seems a bit fragile to me. As without ocamllex, I'd have to use re which isn't available there. The other option is a manual parser, but it will be quite ugly.

**Comment by diml** [comment]
Ok, PR looks good to me, but as said on slack I think we should get  rid of the indexed syntax since it is a bit ad-hoc (it works only for one variable and is not recognized at all in other cases) and we have a better solution almost ready with #950. I suggest to add `first-dep` or `dep0` to replace `dep[0]` in this PR, and we can get rid of it once we merge #950.

**Comment by rgrinberg** [comment]
I think first-dep is better. dep0 implies that depn exists which isn't true.

**Comment by rgrinberg** [comment]
@diml so is the plan to get rid of `first-dep` completely btw? I.e. to make people assign names to things if they want to refer to a particular dep.

**Comment by diml** [comment]
That seems nicer to me. But we can revisit this decision if it's not good in practice.

File: src/super_context.ml [hunk 1]
```diff
@@ -772,23 +801,36 @@ module Action = struct
...
-               Loc.warn loc "Variable '<' used with no explicit \
-                             dependencies@.";
-               [Value.String ""]
-             | dep :: _ ->
-               [Path dep])
-        | "^" -> Some (Value.L.paths deps_written_by_user)
+          if syntax_version < (1, 0) then
+            first_dep ()
+          else
+            Loc.fail loc "Variable '<' is renamed to 'first-dep' in dune"
```
Comment (diml) [inline src/super_context.ml]: The error messages are not consistent: some say `in dune files`, some say `in dune` and some say nothing. `in dune files` seems best to me

Comment (rgrinberg) [reply src/super_context.ml]: Agreed. I made all the errors a little more consistent. Full consistency will probably have to wait for the refactor that makes variable name changes first class.

**Pull request merged**